phoenix-core/src/main/java/org/apache/phoenix/coprocessor/CDCGlobalIndexRegionScanner.java

```diff
@@ -90,128 +90,157 @@ public CDCGlobalIndexRegionScanner(final RegionScanner innerScanner,
+                                }
+                            } else if (cell.getTimestamp() == indexCellTS) {
+                                if (isChangeImageInScope || isPostImageInScope) {
+                                    changeImageObj.put(cdcColumnName,
+                                            this.getColumnValue(cell, cdcColumnInfoList
+                                                    .get(columnListIndex).getColumnType()));
                                 }
                             }
-                            Map<ImmutableBytesPtr, Cell> rowOfCells = new HashMap();
-                            rowOfCells.putAll(rollingRow);
-                            changeTimeline.put(ts, rowOfCells);
                         }
                     }
-
-                    Map<ImmutableBytesPtr, Cell> mapOfCells = changeTimeline.get(indexRowTs);
-                    if (mapOfCells != null) {
-                        Map <String, Object> rowValueMap = new HashMap<>(mapOfCells.size());
-                        for (Map.Entry<ImmutableBytesPtr, Cell> entry: mapOfCells.entrySet()) {
-                            String colName = dataColQualNameMap.get(entry.getKey());
-                            Object colVal = dataColQualTypeMap.get(entry.getKey()).toObject(
-                                    entry.getValue().getValueArray());
-                            rowValueMap.put(colName, colVal);
-                        }
-                        byte[] value =
-                                new Gson().toJson(rowValueMap).getBytes(StandardCharsets.UTF_8);
-                        CellBuilder builder = CellBuilderFactory.create(CellBuilderType.SHALLOW_COPY);
-                        ImmutableBytesPtr family = new ImmutableBytesPtr(firstCell.getFamilyArray(),
-                                firstCell.getFamilyOffset(), firstCell.getFamilyLength());
-                        dataRow = Result.create(Arrays.asList(builder.
-                                setRow(dataRowKey.copyBytesIfNecessary()).
-                                setFamily(family.copyBytesIfNecessary()).
-                                setQualifier(scan.getAttribute((CDC_JSON_COL_QUALIFIER))).
-                                setTimestamp(firstCell.getTimestamp()).
-                                setValue(value).
-                                setType(Cell.Type.Put).
-                                build()));
-                    }
                 }
-                if (dataRow != null && tupleProjector != null) {
-                    IndexUtil.addTupleAsOneCell(result, new ResultTuple(dataRow),
+                Result cdcRow = getCDCImage(preImageObj, changeImageObj, isIndexCellDeleteRow,
+                        indexCellTS, firstCell, isChangeImageInScope, isPreImageInScope,
+                        isPostImageInScope);
+                if (cdcRow != null && tupleProjector != null) {
+                    if (firstCell.getType() == Cell.Type.DeleteFamily) {
```

Comment:
Is it possible to have a Delete Family first cell? Should not be the first cell a put cell?